src/modules/function/declaration.rs

```diff
@@ -98,24 +101,42 @@ impl SyntaxModule<ParserMetadata> for FunctionDeclaration {
                     break
                 }
                 let is_ref = token(meta, "ref").is_ok();
+                let name_token = meta.get_current_token();//saving for possible error
```

Comment:
Usually when a token is saved - it's to save for error propagation. I think we can remove this comment
```suggestion
                let name_token = meta.get_current_token();
```